src/main/webapp/front-end/src/components/Login.jsx

```diff
@@ -27,4 +51,28 @@ export default class Login extends React.Component {
         </form>
       </Grid>
   }
-}
\ No newline at end of file
+}
+
+const mapStateToProps = state => {
+  return {
+      email: state.get('loginEmail').get('email'),
```

Comment:
.get is used for only going 1 level deep into an immutable js object.
.getIn is used for going multiple levels deep:
state.getIn(['loginEmail', 'email'])
also, you can provide a default value by adding a second argument:
state.getIn(['loginEmail', 'email'], null)